From: Tetsu Yamamoto <tyamamot@redhat.com>
Date: Mon, 4 Aug 2008 15:48:20 -0400
Subject: [xen] ia64: revert paravirt to ioremap /proc/pci
Message-id: 20080804194820.8452.2984.sendpatchset@pq0-1.lab.bos.redhat.com
O-Subject: [RHEL5.3 PATCH 5/10] xen-ia64: Revert paravirtualization to ioremap /proc/pci
Bugzilla: 430219
RH-Acked-by: Bill Burns <bburns@redhat.com>

bz430219
# HG changeset patch
# User Alex Williamson <alex.williamson@hp.com>
# Date 1185814315 21600
# Node ID 6d84769b52563c8ecdbe3ee001befcbfbbf6bfcd
# Parent  b0bf9ba32bfe341af07da97d57572659c920fd30
[IA64] Revert paravirtualization to ioremap /proc/pci

Signed-off-by: Jun Kamada <kama@jp.fujitsu.com>

diff --git a/arch/ia64/pci/pci.c b/arch/ia64/pci/pci.c
index d65c412..5c6464a 100644
--- a/arch/ia64/pci/pci.c
+++ b/arch/ia64/pci/pci.c
@@ -615,14 +615,6 @@ pci_mmap_page_range (struct pci_dev *dev, struct vm_area_struct *vma,
 	else
 		vma->vm_page_prot = pgprot_noncached(vma->vm_page_prot);
 
-	if (is_initial_xendomain()) {
-		unsigned long addr = vma->vm_pgoff << PAGE_SHIFT;
-		size_t size = vma->vm_end - vma->vm_start;
-		unsigned long offset = HYPERVISOR_ioremap(addr, size);
-		if (IS_ERR_VALUE(offset))
-			return offset;
-	}
-
 	if (remap_pfn_range(vma, vma->vm_start, vma->vm_pgoff,
 			     vma->vm_end - vma->vm_start, vma->vm_page_prot))
 		return -EAGAIN;
@@ -678,14 +670,6 @@ pci_mmap_legacy_page_range(struct pci_bus *bus, struct vm_area_struct *vma)
 	vma->vm_pgoff += (unsigned long)addr >> PAGE_SHIFT;
 	vma->vm_page_prot = prot;
 
-	if (is_initial_xendomain()) {
-		unsigned long addr = vma->vm_pgoff << PAGE_SHIFT;
-		size_t size = vma->vm_end - vma->vm_start;
-		unsigned long offset = HYPERVISOR_ioremap(addr, size);
-		if (IS_ERR_VALUE(offset))
-			return offset;
-	}
-
 	if (remap_pfn_range(vma, vma->vm_start, vma->vm_pgoff,
 			    size, vma->vm_page_prot))
 		return -EAGAIN;
